src/blocks/block-coauthors/edit.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * React hook that is used to mark the block wrapper element.
+ * It provides all the necessary props like the class name.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/#useblockprops
+ */
+import {
+	BlockControls,
+	BlockContextProvider,
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+	InspectorControls,
+	RichText,
+	__experimentalGetGapCSSValue,
+	AlignmentControl
+} from '@wordpress/block-editor';
+import { TextControl, ToolbarGroup, PanelBody } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { list, grid } from '@wordpress/icons';
+
+import classnames from 'classnames';
+
+import MemoizedCoAuthorTemplateBlockPreview from './components/memoized-coauthor-template-block-preview';
+
+/**
+ * CoAuthor Template Inner Blocks
+ */
+function CoAuthorTemplateInnerBlocks () {
+	return <div { ...useInnerBlocksProps(
+		{ className: 'wp-block-co-authors-plus-coauthor' },
+		{ template : [['co-authors-plus/name']]}
+	) } />;
+}
+
+const ALLOWED_FORMATS = [
+	'core/bold',
+	'core/italic',
+	'core/text-color',
+];
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( { attributes, setAttributes, clientId, context, isSelected } ) {
+
+	const { prefix, separator, lastSeparator, suffix, layout, textAlign } = attributes;
+	const { postId } = context;
+	const authorPlaceholder = useSelect( select => select( 'co-authors-plus/blocks' ).getAuthorPlaceholder(), []);
+	const [ coAuthors, setCoAuthors ] = useState([authorPlaceholder]);
```

Comment:
@douglas-johnson There are a handful of formatting inconsistencies like the spacing in this line, and `useEffect(()=>{` below. Could you run:

```bash
npm run format:js src/blocks
```

to auto-fix formatting on these? Thank you!

Reply:
Yes, addressed in https://github.com/Automattic/Co-Authors-Plus/pull/997/commits/87ef1404bb693a3c7efbd787ff40ccc037aa09d8